hazelcast/src/test/java/com/hazelcast/quorum/QuorumTest.java

```diff
@@ -227,6 +229,49 @@ public boolean apply(Collection<Member> members) {
         }
     }
 
+    @Test
+    public void givenQuorumFunctionConfigured_whenImplementsHazelcastInstanceAware_thenHazelcastInjectsItsInstance() throws Exception {
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(1);
+        String quorumName = randomString();
+        QuorumConfig quorumConfig = new QuorumConfig(quorumName, true);
+        quorumConfig.setQuorumFunctionClassName(HazelcastInstanceAwareQuorumFunction.class.getName());
+
+        Config config = new Config();
+        config.addQuorumConfig(quorumConfig);
+        HazelcastInstance instance = factory.newHazelcastInstance(config);
+
+        assertEquals(instance, HazelcastInstanceAwareQuorumFunction.instance);
+    }
+
+    @Test
+    public void givenQuorumFunctionInstanceConfigured_whenImplementsHazelcastInstanceAware_thenHazelcastInjectsItsInstance() throws Exception {
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(1);
+        String quorumName = randomString();
+        QuorumConfig quorumConfig = new QuorumConfig(quorumName, true);
+        quorumConfig.setQuorumFunctionImplementation(new HazelcastInstanceAwareQuorumFunction());
+
+        Config config = new Config();
+        config.addQuorumConfig(quorumConfig);
+        HazelcastInstance instance = factory.newHazelcastInstance(config);
+
+        assertEquals(instance, HazelcastInstanceAwareQuorumFunction.instance);
+    }
+
+    private static class HazelcastInstanceAwareQuorumFunction implements QuorumFunction, HazelcastInstanceAware {
+        private static volatile HazelcastInstance instance;
+
+        @Override
+        public void setHazelcastInstance(HazelcastInstance instance) {
+            this.instance = instance;
+        }
+
+        @Override
+        public boolean apply(Collection<Member> members) {
+            return false;
+        }
+    }
+
+
```

Comment:
If you have to touch this class, this superfluous newline could also be removed :)